File: CS/XtraPivotGridTutorial.Step6/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrintingLinks;
using DevExpress.XtraPivotGrid;
using DevExpress.XtraCharts;

namespace XtraPivotGridTutorial
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            xtraPivotGridTutorialViewTableAdapter.Fill(nwindDataSet.XtraPivotGridTutorialView);
            pivotGridControl1.BestFit();
            pivotGridControl1.FieldsCustomization(splitContainerControl1.Panel1);
            pivotGridControl1.CollapseAll();
          
        }

        private void pivotGridControl1_ShowingCustomizationForm(object sender, DevExpress.XtraPivotGrid.CustomizationFormShowingEventArgs e)
        {
            e.ParentControl = splitContainerControl1.Panel1;
        }


        private void previewToolStripMenuItem_Click_1(object sender, EventArgs e)
        {
            CompositeLink compositeLink = CreateCompositeLink(pivotGridControl1, chartControl1);
            compositeLink.ShowPreview();
        }

        private void exportToXLSXToolStripMenuItem_Click(object sender, EventArgs e)
        {
            SaveFileDialog saveFileDialog1 = new SaveFileDialog();
            saveFileDialog1.Filter = "Excel Workbook|*.xlsx";
            saveFileDialog1.ShowDialog();
            if (saveFileDialog1.FileName != "")
            {
                CompositeLink compositeLink = CreateCompositeLink(pivotGridControl1, chartControl1);
                compositeLink.ExportToXlsx(saveFileDialog1.FileName);
            }
        }

        CompositeLink CreateCompositeLink(PivotGridControl pivotGridControl, ChartControl chartControl)
        {
            CompositeLink compositeLink = new CompositeLink(new PrintingSystem());
            PrintableComponentLink pivotLink = new PrintableComponentLink();
            pivotLink.Component = pivotGridControl;
            PrintableComponentLink chartLink = new PrintableComponentLink();
            chartLink.Component = chartControl;
            compositeLink.Links.AddRange(new object[] { pivotLink, chartLink });
            return compositeLink;
        }

        

       
    }
}

File: VB/XtraPivotGridTutorial.Step6/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrintingLinks

Namespace XtraPivotGridTutorial
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			xtraPivotGridTutorialViewTableAdapter.Fill(nwindDataSet.XtraPivotGridTutorialView)
			pivotGridControl1.BestFit()
			pivotGridControl1.FieldsCustomization(splitContainerControl1.Panel1)
			pivotGridControl1.CollapseAll()

		End Sub

		Private Sub pivotGridControl1_ShowingCustomizationForm(ByVal sender As Object, ByVal e As DevExpress.XtraPivotGrid.CustomizationFormShowingEventArgs)
			e.ParentControl = splitContainerControl1.Panel1
		End Sub


		Private Sub previewToolStripMenuItem_Click_1(ByVal sender As Object, ByVal e As EventArgs) Handles previewToolStripMenuItem.Click
			compositeLink1.ShowPreview()
		End Sub

		Private Sub exportToXLSXToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs) Handles exportToXLSXToolStripMenuItem.Click
			Dim saveFileDialog1 As New SaveFileDialog()
			saveFileDialog1.Filter = "Excel Workbook|*.xlsx"
			saveFileDialog1.ShowDialog()
			If saveFileDialog1.FileName <> "" Then
				compositeLink1.ExportToXlsx(saveFileDialog1.FileName)
			End If
		End Sub



	End Class
End Namespace
